src/test/ui/nll/borrowed-temporary-error.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
```

Comment:
current output for this test:

```
lunch-box. rustc +nightly borrowed-temporary-error.rs -Znll-dump-cause
error[E0597]: borrowed value does not live long enough
  --> borrowed-temporary-error.rs:20:10
   |
20 |         &(v,)
   |          ^^^^ temporary value does not live long enough
21 |         //~^ ERROR borrowed value does not live long enough [E0597]
22 |     });
   |       - temporary value only lives until here
23 |     println!("{:?}", x);
   |                      - borrow later used here
   |
   = note: borrowed value must be valid for lifetime '_#2r...

error: aborting due to previous error
```

Reply:
this seems pretty good to me, modulo the `borrowed value must be valid for lifetime '_#2r...`, which I know you already removed.

Reply:
I think we should only be showing the `borrowed value must be valid for lifetime '_#2r` for anon lifetimes that cover an entire block and named lifetimes. That shouldn't be that hard to do.

Reply:
I don't understand. I don't think there is any circumstances in which we should show the text `'_#2r`. I guess you are saying that a sentence of this kind only makes sense for either an entire block or named lifetimes?

In that case, I both agree and disagree: it probably makes sense for named lifetimes, but I don't think it makes sense for blocks.

Reply:
Mistyped and you interpreted me correctly. Meant to write `borrowed value must be valid for {}` for the current cases (entire block or named).

For named we currently only point at the signature of the scope that defines it, while for anonymous we point at the entire block on its own span note. We could change this to point at the end where it must live, but that doesn't seem to me entirely different to what we do now, and would lose some context... Of course, this only applies to LL.

I think the best bet is still to continue handling specific cases with more targeted suggestions/explanations, as gaurikholkar was doing. These can get increasingly niche, but those are the cases where `rustc` holding your hand is most impactful and can be a great teachable opportunity.